### ISSUE TITLE
I am trying to run selenium test on the simulator safari to test our mobile web but I get an error message with 0.6.3+ 

### ISSUE THREAD

**Issue by alantouk** [ios-driver/ios-driver#142]
running the following commands gives:

java ios-server-0.6.4.jar -hub http://localhost:4444/grid/register -host localhost -port 5555 -simulators

Error message:

"51:06:616 WARNING AppleLanguage.create  null isn't recognized.Please file a bug on github.You won't be able to start the app in that language."

How do I define the simulator language?

Also is the -simulators flag required? If I don't put that flag then the capabilities appears blank on the grid console...  

Thanks

**Comment by nicegraham** [comment]
you can ignore that message - it's just a warning. 

If you want to run on simulator you will need that flag. 

Assuming with that flag you can see safari as a capability then you should be able to run a test against it.

**Comment by alantouk** [comment]
This is the message from the local grid:

Sep 09, 2013 2:31:30 PM org.openqa.grid.internal.TestSlot getNewSession
INFO: Trying to create a new session on test slot {platform=MAC, simulator=true, browserName=iPhone, maxInstances=1, timeHack=false, variation=Regular, locale=en_GB, CFBundleName=Safari, device=iphone, language=en}

but the test fails because of the following error message:

testPage caused an ERROR: error starting the session: error starting the session: cannot configure simulator Build info: version: 'unknown', revision: 'unknown', time: 'unknown' System info: os.name: 'Mac OS X', os,arch: 'x86_64', os.version: '10.8.4', java.version: '1.7.0_25' Driver info: driver.version: unknown Build info: version: 'unknown', revision: 'unknown'...

**Comment by nicegraham** [comment]
do you have xcode installed? have you launched a simulator manually on the target machine?

Checking the code that error is thrown if ios-driver can't write the simulator settings plist so I assume you'll need to chmod the simulator folder to allow the plist to be written. Possibly:

  sudo chmod 666 /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator*.sdk/Applications/MobileSafari.app/Info.plist

**Comment by alantouk** [comment]
Yes I have xcode installed and can launch a simulator manually. iWebDriver works fine but since that is now deprecated we want to use ios-server.

I have previously ran that chmod command and did it again but still has the same error.

INFO: Available nodes: [host :http://localhost:5555 time out : 300000]

ios-server-0.6.0 used to work fine but now for some reason the capabilities are not being shown so tests can't run...

**Comment by alantouk** [comment]
I think that I have figured this out..

Basically you need to pass in the following DesiredCapabilities:

```
DesiredCapabilities safari = IOSCapabilities.iphone("Safari");
RemoteWebDriver remoteDriver =
          new RemoteWebDriver(remoteAddress, safari);
```

I was previously just passing the browser as iPhone...

**Comment by nirvdrum** [comment]
This should still work using "iPhone" or "iPad" as the browser string.  I haven't tracked down when things broke, but if the capabilities lack a language or locale, the simulator plist file fails to write properly and causes the session creation to fail.  If I null guard the `deviceManager.setL10N` call in `InstrumentsManager#startSession`, everything works as expected.

**Comment by nirvdrum** [comment]
This is should now be fixed in 0.6.5.
